IsraelHiking.Web/src/application/services/route-statistics.service.ts

```diff
@@ -107,22 +105,27 @@ export class RouteStatisticsService {
                 (currentPoint.coordinate[0] - prevPoint.coordinate[0]);
         }
 
-        let pts = resample(routeStatistics.points.map(p=>p.coordinate), { dist: 0.025 }, false);
+	    // calculate total gain & loss:
+        // resample coordinates along route at uniform resolution
+        let coordinates = routeStatistics.points.map(p=>[p.coordinate[0], p.coordinate[1]]);
+        let linterp = linearInterpolator(coordinates);
+        var interpolatedCoordinates = [];
+        for(let x = coordinates[0][0]; x <= coordinates[coordinates.length-1][0]; x+=25) {
+            interpolatedCoordinates.push([x, linterp(x)]);
+        }
+        // filter resampled coordinates to remove sudden jumps
         let median = createMedianFilter(11);
-        let simplifiedCoordinates = pts.map(p => [p[0], median(p[1])]);
-        let previousSimplifiedPoint = simplifiedCoordinates[0];
-        for (let simplifiedPoint of simplifiedCoordinates) {
-            routeStatistics.gain += ((simplifiedPoint[1] - previousSimplifiedPoint[1]) > 0 &&
-                    simplifiedPoint[1] !== 0 &&
-                    previousSimplifiedPoint[1] !== 0)
-                ? (simplifiedPoint[1] - previousSimplifiedPoint[1])
-                : 0;
-            routeStatistics.loss += ((simplifiedPoint[1] - previousSimplifiedPoint[1]) < 0 &&
-                    simplifiedPoint[1] !== 0 &&
-                    previousSimplifiedPoint[1] !== 0)
-                ? (simplifiedPoint[1] - previousSimplifiedPoint[1])
-                : 0;
-            previousSimplifiedPoint = simplifiedPoint;
+        let filteredCoordinates = interpolatedCoordinates.map(p => [p[0], median(p[1])]);
+        // compute total route gain & loss
+        let previousFilteredCoordinate = filteredCoordinates[0];
+        for (let filteredCoordinate of filteredCoordinates) {
+            let elevation_diff = filteredCoordinate[1] - previousFilteredCoordinate[1]
+            if (elevation_diff !=NaN)
```

Comment:
In what case does this receives `NaN`? Also it's better to use `isNaN()` method

Reply:
not sure why, something in my previous changes invoked the need for this (caught it through the tests), but seems that the new changes fixed that too and it can be removed.